### ISSUE TITLE
Add a standalone develop mode for easy develop redash

### ISSUE THREAD

**Issue by zhongjiajie** [getredash/redash#6337]
I checkout the latest code, and when I launch the develop mode to change some code, I only found https://github.com/getredash/redash/wiki/Local-development-setup which develop based on docker-compose, and I think is a little hard for the developer, So I proposed and a standalone mode for the developer.

After that we can launch redash by single command 

```py
redash run standalone
# or maybe need to run wrapper by python, like
python maneger.py --standalone
```

Currently, the main issue is we need to run our service base on redis and postgresql, we need to find some way running in python  alternatives module to hold it

I find some possibility way to solve this:

* [ ] Redis: https://github.com/yahoo/redislite

**Comment by guidopetri** [comment]
@zhongjiajie I don't think we want to support this, given that it'll make it a lot easier to run into "it works on my machine" issues and Docker is more consistent. Why do you think using docker compose is difficult for the developer?

**Comment by justinclift** [comment]
Using docker compose means people need to be familiar with docker (and compose), which is kind of an extra hurdle for starting.

**Comment by guidopetri** [comment]
I mean, sure. But in my view that's equivalent to having to learn how to use an IDE or how to use git in order to develop 😅 maybe the solution is better documentation around how to make and test changes locally?

**Comment by zhongjiajie** [comment]
> @zhongjiajie I don't think we want to support this, given that it'll make it a lot easier to run into "it works on my machine" issues and Docker is more consistent. Why do you think using docker compose is difficult for the developer?

IMO, we can easier develop and debug when all services run in IDE(both pycharm and vscode), and the user can ship the latest code running and check wether it works by seconds, without recreate docker image

**Comment by guidopetri** [comment]
I mean, if you really want to do that, you can still follow the instructions as listed in the Dockerfile exactly. That should install redash on your local as a native application.

Unless you're changing dependencies, recreating the docker image also doesn't take very long - [we copy the code as one of the last steps](https://github.com/getredash/redash/blob/master/Dockerfile#L96-L97) to take advantage of caching, and if you're editing the frontend you'll have to build the frontend locally anyway.

Could you maybe share what your process is, and what bothers you about it/what takes long? Maybe it's just a documentation/process difference?
